pkg/workload/workload.go

```diff
@@ -474,3 +476,40 @@ func HasRetryOrRejectedChecks(wl *kueue.Workload) bool {
 func IsAdmitted(w *kueue.Workload) bool {
 	return apimeta.IsStatusConditionTrue(w.Status.Conditions, kueue.WorkloadAdmitted)
 }
+
+// FindStatusCondition - returns a pointer to the check identified by checkName if found in checks.
+func FindAdmissionCheck(checks []kueue.AdmissionCheckState, checkName string) *kueue.AdmissionCheckState {
```

Comment:
nit: Is there some utility package we could move this function (and the one below to)?

Reply:
They are only used wit workload, creating a new packege for the will just add an extra import wherever used, for now let's just move them in another file and add some unit tests.

Reply:
sgtm